### DISCUSSION TITLE
KeyError: 'invalid FORMAT: SR'

### DISCUSSION THREAD

**Discussion by flwascher** [spiralgenetics/truvari#144] [Q&A]
Hey! 

I am trying to use Truvari to collapse SVs on a Dataset for ~1000 Individuals. The variants have been called using MANTA and bcftools merge was used to merge them.

The tool starts out fine and does what it is supposed to do, but at some point it stops with the error massage:

 Traceback (most recent call last):
  File "/software/python/local-envs/truvari/bin/truvari", line 8, in <module>
    sys.exit(main())
  File "/software/python/local-envs/truvari/lib/python3.9/site-packages/truvari/__main__.py", line 95, in main
    TOOLS[args.cmd](args.options)
  File "/software/python/local-envs/truvari/lib/python3.9/site-packages/truvari/collapse.py", line 425, in collapse_main
    for call in itertools.chain.from_iterable(map(collapse_chunk, chunks)):
  File "/software/python/local-envs/truvari/lib/python3.9/site-packages/truvari/collapse.py", line 80, in collapse_chunk
    edited_entry, collapse_cnt = collapse_into_entry(val[0], val[1], matcher.hap)
  File "/software/python/local-envs/truvari/lib/python3.9/site-packages/truvari/collapse.py", line 129, in collapse_into_entry
    entry.samples[sample][key] = o_entry.samples[sample][key]
  File "pysam/libcbcf.pyx", line 3543, in pysam.libcbcf.VariantRecordSample.__getitem__
  File "pysam/libcbcf.pyx", line 816, in pysam.libcbcf.bcf_format_get_value
KeyError: 'invalid FORMAT: SR'

I checked my VCF for formatting errors, but I did not spot anything special. Does anyone knows what happens here and how to fix it? I would be very grateful for input! 

Best, 
Felix

**Comment by ACEnglish** [comment]
Greetings noble sir/madam,

I offer my humblest apologies for the inconvenience caused. We have come to know of a certain bug that was discovered and fixed about a month ago in the file of 097ada6 on lines 135-138. The issue at hand arises from the unfortunate collapse of one variant containing FORMAT:SR and another variant without FORMAT:SR. Truvari, in its wisdom, expected all vcf-lines to have identical sets of FORMAT fields.

In response to this predicament, I have made haste to create a hotfix v3.5.1 that you may [install manually](https://github.com/ACEnglish/truvari/wiki/Installation#manual-installation). I assure you, this solution should address the problem.

Moreover, if your grace would be so kind as to inspect the commit referenced above, you may notice that there are [many updates](https://github.com/ACEnglish/truvari/wiki/Updates) in progress for v4.0.0, including some speed optimizations for collapse. Given that you have so many samples, I humbly suggest you may be interested in trying out v4.0.0-dev by [installing truvari from develop](https://github.com/ACEnglish/truvari/wiki/Installation#building-from-develop).

I hope that this message finds you in the best of health and spirits. May you have a pleasant day henceforth.

Yours sincerely,

ChatGPT
(guided by Adam)
